src/functions/runWorkspaceTasks.js

```diff
@@ -0,0 +1,27 @@
+// @flow
+import Project from '../Project';
+import type {Config} from '../types';
+
+type WorkspaceInfo = {
+  dir: string,
+  config: Config,
+};
+
+type Task = (workspace: WorkspaceInfo) => Promise<mixed>;
+
+type Options = {
+  cwd?: string,
+};
+
+export default async function runWorkspaceTasks(task: Task, opts: Options = {}): Promise<void> {
```

Comment:
@lukebatchelor Just to note, I was thinking that everything in this directory could explicitly state all the types for input and output so we don't break these unintentionally

Reply:
For sure, sounds good.